### PR TITLE
Implement detector covers

### PR DESCRIPTION
**Pull request by mikerooni**
- [x] Activity
- [x] Energy
- [x] Fluid
- [x] Item
- [x] Maintenance

### CODE REVIEW

**Comment by mikerooni** [comment]
In be3a44e, I've added some slight shadows on the energy detector covers (around the needles specifically) for better visibility against the background, as their colors are the only difference between the regular and advanced energy detector.

Please review that as well. I'm not sure whether the shadow is too much or alright like that.

File: common/src/main/java/com/gregtechceu/gtceu/utils/TextFormattingUtil.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        int index = 0;
+        while (I18n.exists(key + '.' + index)) {
+            keys.add(key + '.' + index);
+            index++;
+        }
+
+        return keys;
+    }
+}
```
Comment (Yefancy) [inline common/src/main/java/com/gregtechceu/gtceu/utils/TextFormattingUtil.java]: why not use the `LangHandler.getMultiLang`

**Review by Yefancy — CHANGES_REQUESTED** [review]

**Comment by Yefancy** [comment]
neat lgtm

File: common/src/main/java/com/gregtechceu/gtceu/api/gui/widget/LongInputWidget.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        this.value = value;
+        onChanged.accept(value);
+
+        return this;
+    }
+
+    private void updateTextFieldRange() {
+        this.textField.setNumbersOnly(min, max);
+        this.setValue(Mth.clamp(this.value, min, max));
+    }
```
Comment (Yefancy) [inline common/src/main/java/com/gregtechceu/gtceu/api/gui/widget/LongInputWidget.java]: merge it with your `IntInputWidget`? thry are similar
a `NumberInputWidget` is better.

File: common/src/main/java/com/gregtechceu/gtceu/common/cover/detector/ActivityDetectorCover.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import net.minecraft.core.Direction;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class ActivityDetectorCover extends DetectorCover {
+    @Getter
+    public static final ManagedFieldHolder MANAGED_FIELD_HOLDER = new ManagedFieldHolder(ActivityDetectorCover.class, DetectorCover.MANAGED_FIELD_HOLDER);
```
Comment (Yefancy) [inline common/src/main/java/com/gregtechceu/gtceu/common/cover/detector/ActivityDetectorCover.java]: never be used, there is no managed fields in this class

File: common/src/main/java/com/gregtechceu/gtceu/common/cover/detector/AdvancedEnergyDetectorCover.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    public ManagedFieldHolder getFieldHolder() {
+        return MANAGED_FIELD_HOLDER;
+    }
+
+    @Persisted @DescSynced @Getter @Setter
+    public long minValue, maxValue;
+    @Persisted @DescSynced @Getter @Setter
+    private int outputAmount;
+    @Persisted @DescSynced @Getter
+    private boolean usePercent;
```
Comment (Yefancy) [inline common/src/main/java/com/gregtechceu/gtceu/common/cover/detector/AdvancedEnergyDetectorCover.java]: do we really need `@DescSynced` here? what are they doing in the remote? 
if its just used for gui, do sync in the gui code is fine.

**Review by Yefancy — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/gregtechceu/gtceu/utils/TextFormattingUtil.java (hunk 1 above)
Comment (mikerooni) [reply common/src/main/java/com/gregtechceu/gtceu/utils/TextFormattingUtil.java]: Ahh, I thought `LangHandler` was only for data *generation*. Didn't see that one.

File: common/src/main/java/com/gregtechceu/gtceu/api/gui/widget/LongInputWidget.java (hunk 2 above)
Comment (mikerooni) [reply common/src/main/java/com/gregtechceu/gtceu/api/gui/widget/LongInputWidget.java]: Hmm, been thinking about a more generic `NumberInputWidget` as well - will try to merge them together.

**Review by Yefancy — APPROVED** [review]

**Pull request merged**